timely/src/dataflow/scopes/child.rs

```diff
@@ -125,7 +125,7 @@ where
         F: FnOnce(&mut Child<Self, T2>) -> R,
     {
         let index = self.subgraph.borrow_mut().allocate_child_id();
-        let path = self.subgraph.borrow().path.clone();
+        let path = self.addr_for_child(index);
```

Comment:
I can believe this change is correct, but I don't understand it.

Reply:
It's hidden in the change to `SubgraphBuilder::new_from`, which doesn't add the index to the path anymore. Maybe it's better to not pass `index` but provide a complete path to said function?

Reply:
I see, so with the change below, rather than pass in `index` and `path`, and have the method append one into the other, we do it ahead of time here with `addr_for_child` and `new_from` just takes the path but uses it in a different sense (previously: "path to `self`" and now: "path to the subscope")?

Reply:
Yup, that's the idea. We could go back the the former implementation, but then might need another allocation for the dataflow root or so.

Reply:
No, this looks better, and having read the call site for `new_from`, it just seems strictly better this way (clearer, even). Just, the process of realizing this was more complicated than the other changes from `Vec<T>` to `Rc<[T]>`. :D